### PR TITLE
Add download figure button

### PR DESCRIPTION
**Pull request by StRhys**
Moves and modifies the download figure button from CMS to chart builder.

This had the side effect of completing #794 on CMS

### CODE REVIEW

File: src/components/chart-preview/ChartPreview.tsx [hunk 1]
```diff
@@ -52,6 +55,42 @@ export const ActualChart = ({
     useSaveCsvData(csv, "chart-data");
   };
 
+  const onFigureDownloadClick = (e: any) => {
+    //TODO rework getting parent node
+    const fullFigureNode =
+      e.target.parentNode.parentNode.parentNode.parentNode.parentNode;
```
Comment (eldeal) [inline src/components/chart-preview/ChartPreview.tsx:61]: Have you tested that this is always a consistent relationship across different chart types? Just wondered if different types might have different nesting.

File: src/components/chart-preview/ChartPreview.tsx [hunk 2]
```diff
@@ -52,6 +55,42 @@ export const ActualChart = ({
...
+
+    let figureCloneNode = fullFigureNode.cloneNode(true);
+    let emptyNode = document.createElement("div");
+    emptyNode.setAttribute(
+      "style",
+      "position: absolute; opacity: 0; pointerEvents: none; width: 1280px;",
+    );
+    figureCloneNode.setAttribute("style", "background-color: white;");
```
Comment (eldeal) [inline src/components/chart-preview/ChartPreview.tsx:69]: Why do we have to set some styles on the emptyNode and some on the figureCloneNode when they end up appended together? Couldn't we do the appends first and all the styles in one go after?

File: src/components/chart-preview/ChartPreview.tsx [hunk 3]
```diff
@@ -52,6 +55,42 @@ export const ActualChart = ({
...
+    emptyNode.append(figureCloneNode);
+    fullFigureNode.append(emptyNode);
```
Comment (eldeal) [inline src/components/chart-preview/ChartPreview.tsx:71]: this append and append back feels confusing to me (but that might be my backend background showing!)

Would it not be simpler to:
```

let emptyNode = document.createElement("div");
emptyNode.append(fullFigureNode.cloneNode(true));
// set all the attributes
fullFigureNode.append(emptyNode);
```
You could potentially avoid figureCloneNode altogether?

Oh hang on, figureCloneNode is what we're downloading. Why do we need to append these new elements to the DOM at all then?

**Review by eldeal — COMMENTED** [review]
Couple of questions, not necessarily any changes. Happy to chat about it if it's easier to explain :)

File: src/components/chart-preview/ChartPreview.tsx (hunk 1 above)
Comment (eldeal) [inline src/components/chart-preview/ChartPreview.tsx:61]: Discussed: try getting to the highest chartbuilder block and do one parentNode for the figure

Comment (StRhys) [reply src/components/chart-preview/ChartPreview.tsx]: added a function to get the top level node of chartBuilder then get the parent parentNode, which will be the full figure.

**Pull request merged**